distribution/pull_v2.go

```diff
@@ -501,6 +519,29 @@ func (p *v2Puller) pullSchema2(ctx context.Context, ref reference.Named, mfst *s
 		return imageID, manifestDigest, nil
 	}
 
+	var descriptors []xfer.DownloadDescriptor
+
+	// Note that the order of this loop is in the direction of bottom-most
+	// to top-most, so that the downloads slice gets ordered correctly.
+	for _, d := range mfst.Layers {
+		layerDescriptor := &v2LayerDescriptor{
+			digest:            d.Digest,
+			repo:              p.repo,
+			repoInfo:          p.repoInfo,
+			V2MetadataService: p.V2MetadataService,
+		}
+
+		if d.MediaType == schema2.MediaTypeForeignLayer && len(d.URLs) > 0 {
+			if runtime.GOOS != "windows" {
```

Comment:
The focus on this PR should be ensuring that decisions made here don't prevent or complicate future additions along these lines. Please help here, as it will make widening the scope of this feature easier in the future.